src/p2p/net_node.inl

```diff
@@ -1361,13 +1363,20 @@ namespace nodetool
     {
       return false;
     }
+    else if (zone.m_current_number_of_out_peers == zone.m_config.m_net_config.max_out_connection_count - 1)
+    {
+      one_conn_below_max = true;
```

Comment:
Shouldn't this be a `<` check (without the `-1`)? Seems like if the connections dropped below the threshold, you want to run the additional logic. I'm just not sure why "one connection below max" is a special case given what you are trying to do.

I know it changes the scaling of when re-connects occur, but why is one less than max a special case (then further discussion about how its implemented)?

Reply:
If it is one connection below max before a connection is made then can assume it is at max connections after the connection is made (later in this function). zone.m_current_number_of_out_peers is not guaranteed to be updated 
directly after the new connection so this is a cheap way to check.

Reply:
On this topic, this patch works better if the tx proxy connection count is not set to a huge value otherwise the reconnect rate is never scaled down. Would you prefer that I implement logic to detect this based on connection count increase over time or just set a hard limit? Lots can go wrong with the first.